dataprofiler/tests/profilers/test_profiler_class_options.py

```diff
@@ -0,0 +1,111 @@
+            option.set({'structured_options.{}.is_enabled'.format(key): False})
+            self.assertFalse(option.structured_options.properties[key].is_enabled)        
+            option.set({'structured_options.{}.is_enabled'.format(key): True})
+            self.assertTrue(option.structured_options.properties[key].is_enabled)        
+    
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option.set({'structured_options.{}.is_enabled.is_enabled'.format(key): True})
+    
+    def test_validate_helper(self):
+        # Valid cases should return [] while invalid cases should return a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Default Configuration Is Valid
+        self.assertEqual([], option._validate_helper())
+        
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            option._validate_helper(1)
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option._validate_helper())))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.structured_options = ProfilerOptions()
+
+        expected_error = set()
+        expected_error.add('{}.structured_options must be a StructuredOptions.'.format(optpth,))
+
+        self.assertSetEqual(expected_error, set(option._validate_helper()))
+            
+    def test_validate(self):
+        # Valid cases should return None while invalid cases should return or throw a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+    
+        # Default Configuration Is Valid
+        self.assertEqual(None, option.validate())
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option.validate(raise_error=False))))
```

Comment:
If raise_error is False, then this is essentially the same as _validate_helper, since it literally just returns the return of the helper. I think it would be more productive to have the test cases for validate raise errors and you verify the right type of error is raised with the right messaging, while the test cases for the helper just check the lists returned.

Reply:
The only problem with this is that I don't want to assume what order the errors will be. If there is a list of errors then all of them are thrown but the regex assertion is only performed on the first line. I could hard code the first error thrown but I felt it was better just to compare the sets that are returned when raise_error is set to False.

Reply:
Alternatively, I can indent this assertion in to the for loop and have it check each error (which would make the raised assertion predictable) - I think this will be a better solution

Reply:
I actually can't do this for this specific case because of the different cardinalities of the sets, but I will do it for the cases where it's only one error being raised.

Reply:
Then is there some way to capture output and turn the string of errors separated by \n into a set and check that way? I understand thats a hassle for verifying a very small extra piece of logic but this test is essentially identical to the validate_helper ones.

Reply:
Took a little bit of tinkering, but yes was able to capture the error output and check that way - pushed that change in the new commit